src/main/java/edu/harvard/iq/dataverse/DatasetField.java

```diff
@@ -407,20 +407,23 @@ public void setValidationMessage(String validationMessage) {
         this.validationMessage = validationMessage;
     }
     
+    
+    // these two booleans are used to wrap around the field type isRequired and isHasChildren methods to also check
+    // if the dataverse has overriden the default values
     @Transient 
     private Boolean required;
+    @Transient 
+    private Boolean hasRequiredChildren;
        
     public boolean isRequired() {
         if (required == null) {
             required = false;
-            if (this.datasetFieldType.isPrimitive() && this.datasetFieldType.isRequired()) {
-                required = true;
-            }
-
-            if (this.datasetFieldType.isHasRequiredChildren()) {
+            
+            if (this.datasetFieldType.isRequired()) {
                 required = true;
             }
```

Comment:
Does it make sense to put a return here?

Reply:
We can't just do that their because if the last if in this method (the one that has the comment "since we don't currently support the use case where the parent is required but no specific children are (the "true/false" case), we override this as false).

That said, it could probably skip some of the code right after by putting it in an else? I don't think it's especially critical, since the other thing that happens is it stores the value in the transient field, so would only get run at most once.